content/resources/how-to-guides/state-changes-across-task-executions.md

```diff
@@ -0,0 +1,171 @@
+
+    | _measurement        | tagKey1           | _field        | _value | _time                |
+    | :------------------ | :---------------- | :------------ | -----: | :------------------- |
+    | example-measurement | example-tag-value | example-field |   30.0 | 2022-01-01T00:00:00Z |
+    | example-measurement | example-tag-value | example-field |   50.0 | 2022-01-01T00:00:00Z |
+
+
+3. Assign states to your data based on thresholds. Store this data in a variable, i.e. “states”. To simplify this example, there are only two states: "ok" and "crit." Store states in the `_level` column (required by the monitor package). 
+
+    ```js
+    states =
+        data
+            |> map(fn: (r) => ({r with _level: if r._value > 40.0 then "crit" else "ok"}))
+    ```
+
+    Where `states` might look like: 
+
+    | _measurement        | tagKey1           | _field        | _value | _level | _time                |
+    | :------------------ | :---------------- | :------------ | -----: | :----- | :------------------- |
+    | example-measurement | example-tag-value | example-field |   30.0 | ok     | 2022-01-01T00:00:00Z |
+    | example-measurement | example-tag-value | example-field |   50.0 | crit   | 2022-01-01T00:01:00Z |
+
+
+4. Write “states” back to InfluxDB. You can write the data to a new measurement or to a new bucket. To write the data to a new measurement, use [`set()`](/flux/v0.x/stdlib/universe/set/) to update the value of the `_measurement` column in your “states” data. 
+
+    ```js
+    states
+        // (Optional) Change the measurement name to write the data to a new measurement
+        |> set(key: "_measurement", value: "new-measurement")
+        |> to(bucket : "example-bucket") 
+    ```
+
+5. Find the latest value from the previous task run and store it in a variable “last_state_previous_task”,
+
+    ```js
+    last_state_previous_task =
+        from(bucket: "example-bucket")
+            |> range(start: date.sub(d: task.every, from: tasks.lastSuccess(orTime: -task.every))
+            |> filter(fn: (r) => r._measurement == "example-measurement")
+            |> filter(fn: (r) => r.tagKey == "example-tag-value")
+            |> filter(fn: (r) => r._field == "example-field")
+            |> last() 
+    ```
+
+    Where `last_state_previous_task` might look like: 
+
+    | _measurement        | tagKey1           | _field        | _value | _level | _time                |
+    | :------------------ | :---------------- | :------------ | -----: | :----- | :------------------- |
+    | example-measurement | example-tag-value | example-field |   55.0 | crit   | 2021-12-31T23:59:00Z |
+
+6. Union “states” and “last_state_previous_task”. Store this data in a variable “unioned_states”.
+
+    ```js
+    unioned_states = union(tables: [states, last_state_previous_task])
+
+    // use the sort() function to guarantee that you preserve the time order
+
+    |> sort(columns: ["_time"], desc: true)
+    ```
```

Comment:
~~~suggestion
6. Union “states” and “last_state_previous_task”. Store this data in a variable “unioned_states”. Use [`sort()`](/flux/v0.x/stdlib/universe/sort/) to ensure rows are ordered by time.

    ```js
    unioned_states =
        union(tables: [states, last_state_previous_task])
            |> sort(columns: ["_time"], desc: true)
    ```
~~~